lib/wallet/src/Cardano/Api/Gen.hs

```diff
@@ -666,9 +668,40 @@ genScriptData =
             oneof
                 [ ScriptDataList <$> vectorOf k smallerGen
                 , ScriptDataMap <$> vectorOf k ((,) <$> smallerGen <*> smallerGen)
-                , ScriptDataConstructor <$> arbitrary <*> vectorOf k smallerGen
+                , ScriptDataConstructor <$> genConstructorIx <*> vectorOf k smallerGen
                 ]
 
+        genConstructorIx = oneof
+            [ choose (0, 10) -- more realistic values
+            , abs <$> arbitrary
```

Comment:
Perhaps:
```suggestion
            , arbitrarySizedNatural
```

Reply:
671a4aec21858983f8b7d5c2e749f086b48520b2

Reply:
I thought `Arbitrary Integer` generated large values, but I see that it doesn't. I replaced it with `arbitrary @Word64` instead.

`arbitrarySizedNatural` does sound like a neat replacement for `choose <small numbers>` though (or went with a mixture of both).